kairos-server/src/state/trie.rs

```diff
@@ -0,0 +1,323 @@
+        let sender_account = self.trie_txn.entry(&sender_hash)?.or_insert_with(|| {
+            unreachable!("sender account should exist");
+        });
+
+        // Remove the amount from the sender's account
+        sender_account.balance = sender_account
+            .balance
+            .checked_sub(transfer.amount)
+            .expect("Sender balance underflow");
+
+        let recipient_account = self.trie_txn.entry(&recipient_hash)?.or_insert_with(|| {
+            tracing::info!("creating new account for recipient");
+            Account::new(transfer.recipient.clone(), 0)
+        });
+
+        // Add the amount to the recipient's account
+        recipient_account.balance = recipient_account
+            .balance
+            .checked_add(transfer.amount)
+            .expect("recipient balance overflow");
+
+        Ok(())
+    }
+
+    fn deposit(&mut self, recipient: &PublicKey, deposit: &Deposit) -> Result<(), AppErr> {
+        tracing::info!("verifying the deposit can be applied");
+
+        let [recipient_hash] = hash_buffers([recipient.as_slice()]);
+
+        let recipient_account = self.trie_txn.entry(&recipient_hash)?.or_insert_with(|| {
+            tracing::info!("creating new account for recipient");
+            Account::new(recipient.clone(), 0)
+        });
+
+        if recipient_account.pubkey != *recipient {
+            return Err(anyhow!("hash collision detected on recipient account").into());
+        }
+
+        recipient_account.balance = recipient_account
+            .balance
+            .checked_add(deposit.amount)
+            .ok_or_else(|| anyhow!("recipient balance overflow"))?;
+
+        Ok(())
+    }
+
+    fn withdraw(&mut self, withdrawer: &PublicKey, withdraw: &Withdraw) -> Result<(), AppErr> {
+        tracing::info!("verifying the withdrawal can be applied");
+
+        let [sender_hash] = hash_buffers([withdrawer.as_slice()]);
+
+        let sender_account = self
+            .trie_txn
+            .get(&sender_hash)?
+            .ok_or_else(|| anyhow!("Sender does not have an account"))?;
+
+        if sender_account.pubkey != *withdrawer {
+            return Err(anyhow!("hash collision detected on sender account").into());
+        }
+
+        if sender_account.balance < withdraw.amount {
+            return Err(anyhow!("sender has insufficient funds").into());
+        }
+
+        let sender_account = self.trie_txn.entry(&sender_hash)?.or_insert_with(|| {
+            unreachable!("sender account should exist");
+        });
+        sender_account.balance = sender_account
+            .balance
+            .checked_sub(withdraw.amount)
+            .expect("sender balance underflow");
+
+        Ok(())
+    }
+}
+
+// We could use a self-referential struct here to avoid Arc, but it's not worth the complexity
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct BatchedTxns {
+    set: HashSet<Arc<Transaction>>,
+    ord: Vec<Arc<Transaction>>,
```

Comment:
Can you document what ord is or better: give it a verboser name. Ordered here seems to refer to the ordered-transactions, that were not already in the set